projects/views.py

```diff
@@ -464,6 +530,36 @@ def edit_project(request):
     return render(request, "projects/edit_project.html", context)
 
 
+@require_POST
+def invite_user(request, project_id):
+    mapper = ProjectAllocationMapper(request)
+    project = mapper.get_project(project_id)
+    if not project_pi_or_admin_or_superuser(request.user, project):
+        messages.error(request, "Only the project PI can invite users.")
+        return InviteUserEmailForm()
+
+    form = InviteUserEmailForm(request.POST)
+    if form.is_valid(request):
+        try:
+            email_address = form.cleaned_data["user_email"]
+            if mapper.email_exists_on_project(project_id, email_address):
```

Comment:
This function is harder to relocate from the mapper. Maybe in a utils.py? Could be used in both views/tasks, is why I say that.